src/inausoft.netCLI.Tests/RegexArgumentDeserializerTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿using inausoft.netCLI.Deserialization;
+using inausoft.netCLI.Tests.Mocks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace inausoft.netCLI.Tests
+{
+    [TestClass]
+    public class RegexArgumentDeserializerTests
+    {
+        [TestMethod]
+        public void RootCommandHandler_RunsProperCommandHandler_ForMultipleOptions()
+        {
+            //Arrange
+            var stringOptionValue = "sampleString";
+            var intOptionValue = 102;
+
+            var args = new string[]
+                            {
+                                "--boolOption",
+                                "--stringOption", stringOptionValue,
+                                "--intOption", intOptionValue.ToString()
+                            };
+
+            var deserializer = new RegexArgumentDeserializer();
+
+            //Act
+            var command = deserializer.Deserialize<Command1>(string.Join(" ", args));
+
+            //Assert
+            Assert.IsTrue(command.BoolOption);
+            Assert.IsNull(command.NotOptionProperty);
+            Assert.AreEqual(stringOptionValue, command.StringOption);
+            Assert.AreEqual(intOptionValue, command.IntOption);
+        }
+
+        [TestMethod]
+        [DataRow("s")]
+        [DataRow(@"C:\Program%Files")]
```

Comment:
This test fails when string contains space inside (ex. `@"C:\Program Files"`. Is it desired behaviour?

Reply:
This is tricky. How to treat --path C:\Progream Files ?? Should Deselializer treat it as a string with spaces in between? I'm open to suggestions.

Reply:
String inside quote could be treated as one phrase, eg. `--path "C:\Program Files" --foo bar`

Reply:
I like generally this approach. But that opens a pandora box potentially. I will create an issue regardin this approach to continue disscusion.

Reply:
#10 was created.